simulators/mujoco_simulator/revolve2/simulators/mujoco_simulator/_custom_mujoco_viewer.py

```diff
@@ -0,0 +1,388 @@
+
+import glfw
+import imageio
+import mujoco
+import mujoco_viewer
+import numpy as np
+import yaml
+
+
+class CustomMujocoViewerMode(Enum):
+    """
+    Enumerate different viewer modes for the CustomMujocoViewer.
+
+    - CLASSIC mode gives an informative interface for regular simulations.
+    - MANUAL mode gives a cut down interface, specific for targeting robot movement manually.
+    """
+
+    CLASSIC = "classic"
+    MANUAL = "manual"
+
+
+class CustomMujocoViewer(mujoco_viewer.MujocoViewer):  # type: ignore
+    """
+    Custom Viewer Object that allows for additional keyboard inputs.
+
+    We need the type ignore since the mujoco_viewer library is not typed properly and therefor the MujocoViewer class cant be resolved.
+    """
+
+    _convex_hull_rendering: bool
+    _transparent: bool
+    _paused: bool
+    _hide_graph: bool
+    _wire_frame: bool
+    _time_per_render: float
+    _loop_count: int
+    _mujoco_version: tuple[int, ...]
+
+    _viewer_mode: CustomMujocoViewerMode
+    _position: int
+
+    def __init__(
+        self,
+        model: mujoco.MjModel,
+        data: mujoco.MjData,
+        start_paused: bool = False,
+        render_every_frame: bool = True,
+        mode: CustomMujocoViewerMode = CustomMujocoViewerMode.CLASSIC,
+    ):
+        """
+        Initialize the Viewer.
+
+        :param model: The mujoco models.
+        :param data: The mujoco data.
+        :param start_paused: If the simulation starts paused or not.
+        :param render_every_frame: If every frame is rendered or not.
+        :param mode: The mode of the viewer (classic, manual).
+        """
+        self._viewer_mode = mode
+        self._position = 0
+        self._paused = start_paused
+        self._render_every_frame = render_every_frame
+        self._mujoco_version = tuple(map(int, mujoco.__version__.split(".")))
+        super().__init__(
+            model,
+            data,
+            mode="window",
+            title="custom-mujoco-viewer",
+            width=None,
+            height=None,
+            hide_menus=False,
+        )
+
+    def _add_overlay(self, gridpos: int, text1: str, text2: str) -> None:
+        """
+        Add overlays.
+
+        :param gridpos: The position on the grid.
+        :param text1: Some text.
+        :param text2: Additional text.
+        """
+        if gridpos not in self._overlay:
+            self._overlay[gridpos] = ["", ""]
+        self._overlay[gridpos][0] += text1 + "\n"
+        self._overlay[gridpos][1] += text2 + "\n"
+
+    def _create_overlay(self) -> None:
+        """Create a Custom Overlay."""
+        topleft = mujoco.mjtGridPos.mjGRID_TOPLEFT
+        # topright = mujoco.mjtGridPos.mjGRID_TOPRIGHT
+        bottomleft = mujoco.mjtGridPos.mjGRID_BOTTOMLEFT
+        # bottomright = mujoco.mjtGridPos.mjGRID_BOTTOMRIGHT
+
+        match self._viewer_mode.value:
+            case "manual":
+                self._add_overlay(topleft, "Iterate position", "[k]")
```

Comment:
```suggestion
                self._add_overlay(topleft, "Iterate position", "[K]")
```

Just matching the rest of the overlay